master/security/icmp-ping.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Use ICMP/ping in policy rules
+---
+
+### Big picture
+
+Use Calico network policy to allow and deny ICMP/ping messages. 
+
+### Value
+
+The **Internet Control Message Protocol (ICMP)** provides valuable network diagnostic functions, but it can also be used maliciously. Attackers can use it to learn about your network, or for DoS attacks. Using Calico network policy, you can control where ICMP is used. For example, you can:
+
+- Allow ICMP ping, but only within your cluster 
+- Allow ICMP for pods launched by operators for diagnostic purposes, but block other uses
+- Temporarily enable ICMP to diagnose a problem, then disable it after the problem is resolved
+- Deny/allow ICMPv4 and/or ICMPv6
+
+### Features
+
+This how-to guide uses the following Calico features:
+
+**GlobalNetworkPolicy** or **NetworkPolicy** with:
+- Positive/negative match criteria for ICMP
+- Protocol match for ICMPv4 and ICMPv6
+- Match on ICMP type and code
+
+### Concepts
+
+#### ICMP packet type and code
+
+Calico network policy also lets you deny and allow specific parts of the ICMP packet for fine-grain control. For example, you can specify ICMP type 5, code 2. For details, see [ICMP type and code](https://en.wikipedia.org/wiki/Internet_Control_Message_Protocol#Control_messages).
```

Comment:
> deny and allow specific parts of the ICMP packet

I think this should say "deny and allow ICMP traffic based on specific types and codes"

The way it is worded now makes it seem like you can allow certain parts of the packet but not others, which doesn't make sense.

Reply:
> For example, you can specify ICMP type 5, code 2

I suggest adding "to match specific ICMP redirect packets", so:

> For example, you can specify ICMP type 5, code 2 to match specific ICMP redirect packets. 

This reads better to me, since it explains why you might want to specify a type/code.

Reply:
Fixed.